src/lib.rs

```diff
@@ -412,61 +427,149 @@ const DERIVE_WHERE_VISITED: &str = "derive_where_visited";
 /// [`Zeroize`]: https://docs.rs/zeroize/latest/zeroize/trait.Zeroize.html
 /// [`ZeroizeOnDrop`]: https://docs.rs/zeroize/latest/zeroize/trait.ZeroizeOnDrop.html
 #[proc_macro_attribute]
-#[cfg_attr(feature = "nightly", allow_internal_unstable(core_intrinsics))]
 pub fn derive_where(
 	attr: proc_macro::TokenStream,
 	original_input: proc_macro::TokenStream,
 ) -> proc_macro::TokenStream {
-	let attr: TokenStream = attr.into();
-	let input: TokenStream = original_input.clone().into();
-	let input = quote! {
-		#[derive_where(#attr)]
-		#input
-	};
+	let attr = TokenStream::from(attr);
+	let mut original_input = TokenStream::from(original_input);
+	let mut input = quote_spanned! { attr.span()=> #[derive_where(#attr)] };
+	input.extend(original_input.clone());
 
+	match syn::parse2::<DeriveInput>(input) {
+		Ok(input) => match derive_where_internal(input.clone()) {
+			Ok(item) => item.into(),
+			Err(error) => {
+				let mut clean_input =
+					input_without_derive_where_attributes(input).into_token_stream();
+				clean_input.extend(error.into_compile_error());
+				clean_input.into()
+			}
+		},
+		Err(error) => {
+			original_input.extend(error.into_compile_error());
+			original_input.into()
+		}
+	}
+}
+
+/// Convenient way to deal with [`Result`] for [`derive_where`].
```

Comment:
Apparently this `Result` needs to be explicitly linked

Reply:
Done.